src/kernels/kernel.ts

```diff
@@ -340,11 +358,24 @@ export class Kernel implements IKernel {
             );
         }
     }
-    private async startNotebook(options: { disableUI?: boolean } = { disableUI: false }): Promise<INotebook> {
+    private async startNotebook(options: IDisplayOptions = new DisplayOptions(false)): Promise<INotebook> {
+        if (this.notebookDocument.isClosed) {
+            traceWarning(`Notebook closed`);
+            throw new CancellationError();
+        }
         this._startedAtLeastOnce = true;
+        traceInfoIfCI(
+            `Start Notebook (options.disableUI=${options.disableUI}) for ${getDisplayPath(this.notebookDocument.uri)}.`
+        );
+        console.error(new Error('(startNotebook). Where am I called from'));
         if (!options.disableUI) {
             this.startupUI.disableUI = false;
         }
+        options.onDidChangeDisableUI(() => {
```

Comment:
This was missing, if we have auto start, then we don't display, but when running cells, the errors are not displayed as we're using the stale state of hiding the UI.